src/main/java/gov/cabinetofice/gapuserservice/service/encryption/AESService.java

```diff
@@ -0,0 +1,125 @@
+            response.setIV(initVector);
+            response.setPrivateKey(secretKey);
+            response = signingService.signMessage(response);
+            response.setShaSum(sha512Service.getSHA512SecurePassword(response.getEncryptedText() + response.getSignature()));
+            return Optional.of(response);
+        }
+        catch (NoSuchAlgorithmException
+                | InvalidKeyException
+                | InvalidAlgorithmParameterException
+                | NoSuchPaddingException
+                | BadPaddingException
+                | IllegalBlockSizeException e) {
+            e.printStackTrace();
+        }
+        return Optional.empty();
+    }
+
+    private void GenerateInitVector(byte[] initVector) {
+        SecureRandom secureRandom = new SecureRandom();
+        secureRandom.nextBytes(initVector);
+    }
+
+    public Optional<String> decrypt(EncryptedResponseDto response) {
+        var shaSum  = response.getEncryptedText() + response.getSignature();
+        shaSum = sha512Service.getSHA512SecurePassword(shaSum);
+
+        if(signingService.verify(response) && response.getShaSum().equals(shaSum)) {
+            try {
+                Cipher cipher = Cipher.getInstance(CIPHER_SELECTOR);
+                SecretKeySpec spec = new SecretKeySpec(secretKey.getEncoded(), KEY_SELECTOR);
+                GCMParameterSpec gcmSpec = new GCMParameterSpec(GCM_TAG_LENGTH * 8, response.getIV());
+                cipher.init(Cipher.DECRYPT_MODE, spec, gcmSpec);
+                return Optional.of(new String(cipher.doFinal(DatatypeConverter.parseHexBinary(response.getEncryptedText()))));
+            }
+            catch (NoSuchAlgorithmException
+                    | InvalidKeyException
+                    | InvalidAlgorithmParameterException
+                    | NoSuchPaddingException
+                    | BadPaddingException
+                    | IllegalBlockSizeException e) {
+                e.printStackTrace();
+            }
+        }
+        else {
+            logger.error("Failed to verify signature");
+        }
+        return Optional.empty();
+    }
+
+    public Optional<String> decrypt(final String message, final byte[] iv) {
+
+        try {
+            Cipher cipher = Cipher.getInstance(CIPHER_SELECTOR);
+            SecretKeySpec spec = new SecretKeySpec(secretKey.getEncoded(), KEY_SELECTOR);
+            GCMParameterSpec gcmSpec = new GCMParameterSpec(GCM_TAG_LENGTH * 8, iv);
+            cipher.init(Cipher.DECRYPT_MODE, spec, gcmSpec);
+            return Optional.of(new String(cipher.doFinal(DatatypeConverter.parseHexBinary(message))));
+        } catch (NoSuchAlgorithmException
+                | InvalidKeyException
+                | InvalidAlgorithmParameterException
+                | NoSuchPaddingException
+                | BadPaddingException
+                | IllegalBlockSizeException e) {
+            e.printStackTrace();
```

Comment:
Should catch this exception and log it using the Logger mechanism rather than just dumping the stack trace